netket/dynamics.py

```diff
@@ -1 +1,362 @@
+            samples_r, eloc_r / self._n_samples, self._grads
+        )
+
+        self._grads = tree_map(_sum_inplace, self._grads)
+        self._grads = tree_map(lambda x: -1.0j * x, self._grads)
+
+        self._dp = self._sr.compute_update_onthefly(samples_r, self._grads, self._dp)
+
+    else:
+        # When using the SR (Natural gradient) we need to have the full jacobian
+        self._grads, self._jac = self._machine.vector_jacobian_prod(
+            samples_r,
+            eloc_r / self._n_samples,
+            self._grads,
+            return_jacobian=True,
+        )
+
+        self._grads = tree_map(_sum_inplace, self._grads)
+        self._grads = tree_map(lambda x: -1.0j * x, self._grads)
+
+        self._dp = self._sr.compute_update(self._jac, self._grads, self._dp)
+
+    return self._dp
+
+
+@_time_evo.register(SteadyState)
+def _time_evo_vmc(self):
+    """
+    Computes the derivative of the time-dependent ODE for
+    a SteadyState (lindblad) driver.
+
+    """
+
+    self._obs_samples_valid = False
+    self._sampler.reset()
+    self._sampler.generate_samples(self._n_discard)
+    self._samples = self._sampler.generate_samples(
+        self._n_samples_node, samples=self._samples
+    )
+    self._lloc, self._loss_stats = self._get_mc_superop_stats(self._lind)
+    self._loss1_stats = _statistics(self._lloc.T)
+    lloc = self._lloc - _mean(self._lloc)
+    samples_r = self._samples.reshape((-1, self._samples.shape[-1]))
+    lloc_r = lloc.reshape(-1, 1)
+    self._grads, self._jac = self._machine.vector_jacobian_prod(
+        samples_r, lloc_r / self._n_samples, self._grads, return_jacobian=True
+    )
+    self._grads = tree_map(_sum_inplace, self._grads)
+    self._dp = self._sr.compute_update(self._jac, self._grads, self._dp)
+    return self._dp
+
+
+def create_timevo(op, sampler, *args, julia=False, **kwargs):
+    """
+    Constructs the correct driver given the forward-time evolution
+    generator `op`, the sampler `sampler` and other arguments.
+
+        This function also constructs the function computing the function
+        (or derivative) defining the ode, and returns it.
+
+        If `julia=False`, then the function has signature fun(time, weights),
+        otherwise it has signature fun(weights, params, time) and the second
+        argument is ignored.
+        This option is thought to be used with integrators from
+        diffeq.py
+    """
+
+    machine = sampler._machine
+    if not isinstance(machine, AbstractDensityMatrix):
+        driver = Vmc(op, sampler, *args, optimizer=None, **kwargs)
+    else:
+        driver = SteadyState(op, sampler, *args, optimizer=None, **kwargs)
+
+    flatten = machine.numpy_flatten
+    unflatten = lambda w: machine.numpy_unflatten(w, machine.parameters)
+
+    def _fun(t, w):
+        machine.parameters = unflatten(w)
+        dp = _time_evo(driver)
+        return flatten(dp)
+
+    if julia:
+
+        def fun(u, p, t):
+            _fun(t, w)
```

Comment:
This looks like there is a mixup between `u` and `w` here, right?
```suggestion
    if julia:
        def fun(u, p, t):
            _fun(t, u)
```